vulcan/lib/client/jsx/components/workflow/user_interactions/inputs/scDGE.tsx

```diff
@@ -48,54 +50,60 @@ export default function DiffExpSC({
     return opts;
   }, [data])
   
-  const [doSubset, setSubset] = useState(false);
-  const addSubset = (vals = {...value}) => {
-    vals['subset_meta'] = null
-    vals['subset_use'] = null
-    onChange(some(vals));
-  }
-  const removeSubset = (vals = {...value}) => {
-    if (Object.keys(vals).includes('subset_meta')) delete vals['subset_meta']
-    if (Object.keys(vals).includes('subset_use')) delete vals['subset_use']
-    onChange(some(vals));
-  }
-  function toggleSubsetting(subset: boolean) {
-    if (!subset) removeSubset();
-    if (subset) addSubset();
-    setSubset(subset)
-  }
-  
-  const setDEMethod = (method: string, vals = {...value}) => {
-    let newVals = {...output_sets[method]}
-    if (doSubset) {
-      newVals['subset_meta'] = vals['subset_meta']
-      newVals['subset_use'] = vals['subset_use']
-    }
+  const setDEMethod = (label: string, x: string, label_tags = method_labels, vals = {...value}) => {
+    // x = filler for 'key' provided by input piece which will not match the current aims.
+    let newVals = {...output_sets[label_tags[label]]}
+    newVals['subset'] = vals['subset']
     onChange(some(newVals));
   }
   
   const updateValue = (newValue: any, key: string, prevValues = {...value}) => {
     prevValues[key] = newValue;
     onChange(some(prevValues));
   };
-  
+
   const warn =
     <div>
       <p></p>
       IMPORTANT NOTE: The scanpy DE algorithm cannot correct for batch effects. We are currently investigating alternatives to add such functionality.
       <p></p>
     </div>
 
+  // Bringing pre-subsetDataFrameInput value-structure up-to-date.
+  let SubsetComps = null;
+  if (value) {
+    if (value['subset']) {
+    SubsetComps = subsetDataFrameInput(
+      "subset", updateValue, value['subset'], "Step 2: Subset by features to focus on certain cells? ",
+      {...allData}, false, "secondary")
+    } else {
+      const updateOldSubset = () => {
+        let new_subset = ( !(value['subset_meta'] && value['subset_use'])) ? {} : {
+          'methods': [ [value['subset_meta']].concat(value['subset_use']) ],
+          'logic': [[]]
+        }
+        let prevValues = {...value}
+        delete prevValues['subset_meta']
+        delete prevValues['subset_use']
+        updateValue(new_subset, 'subset', prevValues)
+      }
+      SubsetComps = 
+        <Button
+          color={"primary"}
+          onClick={(x) => {updateOldSubset()}}>
+          Step 2: Subset... UPDATE NEEDED. Click to update.
+        </Button>
+    }
+  }
+
```

Comment:
I don't love that this ultimately brings back half the amount of code which the `subsetDataFrameInput` piece allowed me to remove, but addresses https://github.com/mountetna/monoetna/pull/746#issuecomment-1036517692

Reply:
![scDGE_update](https://user-images.githubusercontent.com/38870347/153674014-5aee5065-ca37-45b4-93e8-1969bc0d026a.gif)